transaction/core/src/blockchain/quorum_set.rs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use alloc::{
+    string::{String, ToString},
+    vec,
+    vec::Vec,
+};
+use core::{
+    hash::{Hash, Hasher},
+    str::FromStr,
+};
+use mc_common::{NodeID, ResponderId};
+use mc_crypto_digestible::Digestible;
+use mc_crypto_keys::Ed25519Public;
+use prost::{Message, Oneof};
+use serde::{Deserialize, Serialize};
+
+/// The quorum set defining the trusted set of peers.
+#[derive(Clone, Deserialize, Digestible, Message, Ord, PartialOrd, Serialize)]
+pub struct QuorumSet {
```

Comment:
I opted to define a minimal/simple `QuorumSet` after grappling with `scp::QuorumSet`, trying to make it implement `Message`, which `LedgerDB` requires to serialize values. Note that this PR also includes `impl [Try]From` to convert back and forth between them.

Reply:
Making `GenericNodeId` include `prost::Message` proved problematic -- `mc_common::NodeID` and `mc_common::ResponderId` would also need to implement `Message`, and we don't currently have a `prost` dependency in `common`.

The other main issue was the impedance mismatch between how Prost handles `oneof` (hence `mc_transaction_core::QuorumSetMemberWrapper`) and how `scp::QuorumSet` is structured. I don't remember exactly what I got stuck on there, but I do remember it feeling gnarly, and giving up on that approach.